### DISCUSSION TITLE
How to resolve Node.js core modules

### DISCUSSION THREAD

**Discussion by codegenic-works** [modernweb-dev/web#1393] [Q&A]
Hello! I recently installed a npm package that makes use of some of the Node.js core modules like `fs`, `os`, and `crypto`. When trying to run the web server I get errors stating that those imports cannot be resolved. I was looking through the documents and I might be missing it, but, is there a setting or plugin that I need to use to polyfill those?

Just to cover the basis, here is my command line:
`wds --app-index index.html --open --watch --node-resolve`

And my web-dev-server.config.js is:
```javascript
const rollupReplace = require('@rollup/plugin-replace');
const { fromRollup } = require('@web/dev-server-rollup');
const envConfig = require('./dev.config.js');

const replace = fromRollup(rollupReplace);

module.exports = {
    plugins: [
        replace({
            ...envConfig ,
            exclude: 'node_modules/**',
            preventAssignment: true
        })
    ]
};
```

Much thanks in advance!

**Comment by daKmoR** [comment]
web dev server is executing test in an actual browser... and a browser does not have `fs` nor does it have access to the filesystem...

Are you maybe looking for a node testing tool like mocha or uvu?

**Comment by electrovir** [comment]
Add this to your config:

```javascript
import nodePolyfills from 'rollup-plugin-node-polyfills';
import {fromRollup} from '@web/dev-server-rollup';

export default {
    plugins: [
        fromRollup(nodePolyfills)(),
    ],
}
```

**Reply by shinichy** [reply]
Does `rollup-plugin-node-polyfills` work? I tried it, but I got the following error.

```
Error while handling server request.
[Error: ENOENT: no such file or directory, stat '/__web-dev-server__/rollup/polyfill-node:os.js'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'stat',
  path: '/__web-dev-server__/rollup/polyfill-node:os.js',
  expose: true,
  statusCode: 404,
  status: 404
}
```

**Reply by shinichy** [reply]
Never mind. I found it doesn't work well with `"@rollup/plugin-commonjs"` plugin.
